src/file_pipe_log/pipe_builder.rs

```diff
@@ -100,6 +99,20 @@ impl<F: FileSystem> DualPipesBuilder<F> {
     /// Scans for all log files under the working directory. The directory will
     /// be created if not exists.
     pub fn scan(&mut self) -> Result<()> {
```

Comment:
No need for a closure to accommodate one single corner case. Maybe use a read_only flag too.

Reply:
`read_only` can't describe the behavior accurately. The last log file in the queue is still opened in read-write mode.

Reply:
The default behavior is to open active in rw, others in r. The corner case behavior is to open all files in r. You only needs a flag to describe the corner case behavior, namely "read_only" or "force_read_only".